harvard-university-for-the-creative-arts.csl

```diff
@@ -167,20 +198,25 @@
           </group>
         </group>
       </else-if>
-      <else-if type="map" match="any">
-        <group delimiter=" ">
-          <text variable="genre" prefix="[" suffix="]"/>
-          <text variable="scale" suffix="."/>
-          <text variable="publisher-place" suffix=":"/>
-          <text variable="publisher" suffix="."/>
-          <text variable="collection-title" prefix="(" suffix=")"/>
-        </group>
-      </else-if>
       <else>
-        <group delimiter=": ">
-          <text variable="publisher-place"/>
-          <text variable="publisher" suffix="."/>
-        </group>
+        <choose>
+          <if match="any" variable="publisher" type="motion_picture broadcast report musical_score song entry-dictionary entry-encyclopedia">
+            <text variable="publisher" suffix="."/>
+          </if>
+          <else>
+            <text value="s.n." prefix="(" suffix=")."/>
+          </else>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-place">
+    <choose>
+      <if match="any" variable="publisher-place" type="broadcast motion_picture report musical_score song entry-dictionary entry-encyclopedia manuscript">
+        <text variable="publisher-place"/>
+      </if>
+      <else>
+        <text value="s.l." prefix="(" suffix=")"/>
```

Comment:
> @adam3smith, I haven't looked, but I don't think we have a ticket to add "sine loco" and "sine nomine" as CSL terms, but it probably makes sense to add them. Agreed?

@adam3smith, bump. Only ticket I could find is https://github.com/citation-style-language/schema/issues/90#issuecomment-4329346, and there are at least a bunch of styles that that use the "sine loco" (https://github.com/search?q=org%3Acitation-style-language+loco&type=Code) and "sine nomine" (https://github.com/search?utf8=%E2%9C%93&q=org%3Acitation-style-language+nomine&type=Code).